python/mxnet/ndarray/op.py

```diff
@@ -83,16 +91,29 @@ def _make_ndarray_function(handle, name):
                          ret_type)
 
     dtype_name = None
+    shape_name = None
+    ctx_name = None
     arr_name = None
     ndsignature = []
     signature = []
     ndarg_names = []
     kwarg_names = []
+    is_random_op = _is_random_op(func_name)
     for i in range(narg):
         name, atype = arg_names[i], arg_types[i]
         if name == 'dtype':
             dtype_name = name
-            signature.append('%s=_Null'%name)
+            signature.append('%s=_Null' % name)
+        elif name == 'shape' and is_random_op:
```

Comment:
Just change the default value to TShape(1) in the backend

Reply:
ok I see the problem here. still don't quite like the is_random_op solution though.

Reply:
agree it is a little bit hacky, but let merge a runnable version first

Reply:
Ok. I will submit another PR making the changes in the way as @piiswrong suggested after my another work is finished.